tests/sets/t15435.nim

```diff
@@ -0,0 +1,41 @@
+discard """
+errormsg: "type mismatch: got <set[uint8], set[range 1..5(uint8)]>"
+nimout: '''t15435.nim(38, 13) Error: type mismatch: got <set[uint8], set[range 1..5(uint8)]>
+but expected one of:
+proc `<`[T](x, y: set[T]): bool
+  first type mismatch at position: 2
+  required type for y: set[T]
+  but expression 'x' is of type: set[range 1..5(uint8)]
+20 other mismatching symbols have been suppressed; compile with --showAllMismatches:on to see them
+
+expression: {1'u8, 5} < x'''
+"""
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+## line 30
```

Comment:
Is it essential for the test that the code starts at line 30?

Reply:
No, I arbitrarily picked a rounded number. 

The discard block at the top has line numbers and they need to match exactly. I am following a convention which I have seen in many similar tests that check again the printed error and stdout.

Reply:
I don't see how starting at line 30 has any advantage here, you copy paste the line number in the error message anyway. I have never seen a convention of 17 blank lines in any test case...

Reply:
> I have never seen a convention of 17 blank lines in any test case...

It's not about the "17 lines". The discard block are the actual tests are usually spaced out so that if more tests are added at the end, and the discard block needs to extend, then you don't have to move the original test and refactor the line numbers in the original discard block.

Reply:
Here's a crude search of other tests that have this style of tests:

![image](https://user-images.githubusercontent.com/3578197/139608167-5917dc10-f3fc-4b53-9fab-11377b59e5b9.png)